Core/Spells: Implement random spell attributes

tisk opened the pull request.

jfb-star committed:
Core/Spells: Implement SPELL_ATTR1_THREAT_ONLY_ON_MISS

jfb-star committed:
Core/Spells: Implement SPELL_ATTR5_NO_PARTIAL_RESISTS

jfb-star committed:
Core/Spells: Implement SPELL_ATTR2_NO_SCHOOL_IMMUNITIES

jfb-star committed:
Core/Spells: Implement SPELL_ATTR8_USE_TARGET_LEVEL_FOR_SPELL_SCALING

jfb-star committed:
Core/Spells: Implement SPELL_ATTR8_REMOVE_OUTSIDE_DUNGEONS_AND_RAIDS

Shauren commented:
I will make an exception this time but please do not touch attributes from enums I have not yet processed (6 and up) - this interferes with my own work

jfb-star committed:
Core/Spells: Fix reseting swing timer logic
- Implemented SPELL_ATTR7_RESET_SWING_TIMER_AT_SPELL_START
- Fixed SPELL_ATTR6_DOESNT_RESET_SWING_TIMER_IF_INSTANT

tisk commented:
> I will make an exception this time but please do not touch attributes from enums I have not yet processed (6 and up) - this interferes with my own work

Ok then i'll wait for the others :)

mdX7 commented on src/server/game/Miscellaneous/SharedDefines.h:
```diff
@@ -679,7 +679,7 @@ enum SpellAttr7 : uint32
     SPELL_ATTR7_SILENCE_ONLY_NONPLAYER           = 0x00001000, // TITLE Can Cause Silence
     SPELL_ATTR7_CAN_ALWAYS_BE_INTERRUPTED        = 0x00002000, // TITLE No UI Not Interruptible DESCRIPTION Can always be interrupted, even if caster is immune
     SPELL_ATTR7_UNK14                            = 0x00004000, // TITLE Unknown attribute 14@Attr7
-    SPELL_ATTR7_UNK15                            = 0x00008000, // TITLE Unknown attribute 15@Attr7 DESCRIPTION Exorcism - guaranteed crit vs families?
```
the comment on this line needs a certain syntax: `TITLE <title> DESCRIPTION <description>` which will then be parsed by https://github.com/TrinityCore/TrinityCore/blob/master/contrib/enumutils_describe.py to generate the file `enuminfo_SharedDefines.cpp`

Shauren replied:
@mdX7 doesnt matter - SharedDefines changes wont be merged (i plan to tear apart this PR and push it in parts)
